### DISCUSSION TITLE
What UDP port should I allow in firewall to make searx search doesn't timeout?

### DISCUSSION THREAD

**Discussion by DitaAjiPratama** [searx/searx#3235] [Installation issue: Official wiki] [answered]
When I disallow All UDP on All Ports in my firewall configuration, it make my search result have a timeout. I use installation on NginX.

Here is the result:
![2022-05-15-155520_760x130_scrot](https://user-images.githubusercontent.com/26424128/168465812-655c2ae3-a641-4df4-b9cf-e3ac74634a1a.png)

**Comment by unixfox** [comment]
Searx only use TCP ports so allow all ports TCP outgoing.

**Reply by DitaAjiPratama** [reply]
Here is my outbound rules:
TCP 80
TCP 443
UDP All Ports

And it works like a charm.

But when I disallow just UDP All Ports, it will turn like this:
![2022-05-15-155520_760x130_scrot](https://user-images.githubusercontent.com/26424128/168464961-635eee96-c256-4baa-812e-1045515cb9be.png)

**Reply by unixfox** [reply]
It's probably an issue on your setup. Searx doesn't use UDP ports.

**Reply by dalf** [reply]
Just in case: Check the DNS configuration.
DNS errors are reported as timeout errors.

**Comment by DitaAjiPratama** [comment, marked as answer]
I tried to allow the UDP DNS port (53) and now it's working fine.

Here is my full firewall setting:

Inbound
TCP 80 for HTTP
TCP 443 for HTTPS

Outbound
TCP 80 for HTTP
TCP 443 for HTTPS
UDP 53 for DNS

but I still have no idea why I must allow UDP 53 Outbound to make it works.

**Reply by unixfox** [reply]
Maybe it's because your setup use an external DNS server and it needs the port 53 UDP opened.

Every component on your system that needs to do name resolution will need to have this port opened, not just searx.

I would even say that it's useless to filter the outgoing ports because that doesn't really provide any security because if a bad program wants to connect to another server it will just use the common ports like 80 or 443. Only filtering inbound is important.

**Reply by DitaAjiPratama** [reply]
Thank you for the tips. I'm not really good at security.
